experiments/storm/src/main/scala/io/gearpump/experiments/storm/producer/StormProducer.scala

```diff
@@ -45,4 +53,17 @@ private[storm] class StormProducer(gearpumpSpout: GearpumpSpout,
     gearpumpSpout.next(msg)
     self ! Message("continue")
   }
+
+  override def receiveUnManagedMessage: Receive = {
+    case CheckpointClock(optClock) =>
+      optClock.foreach { clock =>
+        gearpumpSpout.onCheckpointClock(clock)
+      }
+      getCheckpointClock
+  }
+
+  def getCheckpointClock: Unit = {
```

Comment:
The name is very confusing. It implies me that the method will return a checkpoint clock, but actuall it checks checkpoints internally.